test/mocks/v2alpha1/testing_snapshot.sk.go

```diff
@@ -86,6 +88,82 @@ func (s TestingSnapshot) HashFields() []zap.Field {
 	return append(fields, zap.Uint64("snapshotHash", snapshotHash))
 }
 
+func (s *TestingSnapshot) GetResourcesList(resource resources.Resource) (resources.ResourceList, error) {
```

Comment:
How does this API differ from using s.Mock.AsResources directly?

Reply:
When you don't know the type of `resource`.